cloud/blockstore/libs/diagnostics/profile_log_ut.cpp

```diff
@@ -388,6 +392,27 @@ Y_UNIT_TEST_SUITE(TProfileLogTest)
             env.EventProcessor.FlatMessages[0]
         );
     }
+
+    Y_UNIT_TEST(TestFlushOnDestruct)
+    {
+        TEnv env;
+        env.ProfileLog->Write(
+            {"disk2",
+             TInstant::Seconds(3),
+             IProfileLog::TReadWriteRequest{
+                 EBlockStoreRequest::WriteBlocks,
+                 TDuration::MilliSeconds(300),
+                 TDuration::MilliSeconds(42),
+                 TBlockRange64::WithLength(10, 20),
+             }});
+        env.Scheduler = nullptr;
```

Comment:
to make test more readable and explicitly state that you want to process log without running the scheduler you can change ProcessLog signature (instead of setting env.Scheduler to nullptr and then check it)
```c++
void ProcessLog(bool runScheduler =  true)
```